proof/verifier.go

```diff
@@ -169,9 +169,52 @@ func verifyTaprootProof(anchor *wire.MsgTx, proof *TaprootProof,
 
 // verifyInclusionProof verifies the InclusionProof is valid.
 func (p *Proof) verifyInclusionProof() (*commitment.TapCommitment, error) {
-	return verifyTaprootProof(
-		&p.AnchorTx, &p.InclusionProof, &p.Asset, true,
+	// Determine if we're dealing with v0 or v1 proofs.
+	hasV1Proofs := p.InclusionProof.CommitmentProof != nil &&
+		len(p.InclusionProof.CommitmentProof.STXOProofs) > 0
+
+	if !hasV1Proofs {
+		return verifyTaprootProof(
+			&p.AnchorTx, &p.InclusionProof, &p.Asset, true,
+		)
+	}
+
+	commitVersions := make(map[uint32][]commitment.TapCommitmentVersion)
+	p2trOutputs := make(P2TROutputsSTXOs)
+	outIdx := p.InclusionProof.OutputIndex
+	p2trOutputs[outIdx] = make(fn.Set[asset.SerializedKey])
+
+	// Collect the STXOs from the new asset.
+	stxoAssets, err := asset.CollectSTXO(&p.Asset)
+	if err != nil {
+		return nil, fmt.Errorf("error collecting STXO assets: %w", err)
+	}
+
+	// Map STXOs by serialized key.
+	assetMap := make(map[asset.SerializedKey]*asset.Asset)
+	for idx := range stxoAssets {
+		stxoAsset := stxoAssets[idx].(*asset.Asset)
+		key := asset.ToSerialized(stxoAsset.ScriptKey.PubKey)
+		assetMap[key] = stxoAsset
+		p2trOutputs[outIdx].Add(key)
+	}
+
+	baseProof := p.InclusionProof
+
+	commitment, err := verifySTXOProofSet(
+		&p.AnchorTx, baseProof, assetMap, p2trOutputs, commitVersions,
+		true,
 	)
+	if err != nil {
+		return nil, err
+	}
+
+	err = p.verifyRemainingOutputs(p2trOutputs)
```

Comment:
I don't think we need to call into this. We only added a single output index to the map and we expect that to be empty now, after the STXO verification. So we just need to check for map emptiness.